packages/ui-components/src/components/NativeSelect/NativeSelect.stories.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from "react"
+import { Meta, StoryFn } from "@storybook/react"
+
+import { NativeSelect, NativeSelectProps } from "./NativeSelect.component"
+import { NativeSelectOption } from "../NativeSelectOption/NativeSelectOption.component"
+import { NativeSelectOptionGroup } from "../NativeSelectOptionGroup/NativeSelectOptionGroup.component"
+
+export default {
+  title: "Forms/NativeSelect/NativeSelect",
+  component: NativeSelect,
+  argTypes: {
+    options: {
+      table: {
+        disable: true,
+      },
+    },
+    children: {
+      control: false,
+    },
+  },
+} as Meta<typeof NativeSelect>
+
+const Template: StoryFn<NativeSelectProps> = (args) => <NativeSelect {...args} />
+
+export const Default = Template.bind({})
+Default.args = {
+  children: [
+    <NativeSelectOption key={0} value="o-1" label="Option 1" />,
+    <NativeSelectOption key={1} value="o-2" label="Option 2" />,
+    <NativeSelectOption key={2} value="o-3" label="Option 3" />,
+  ],
+}
+
+export const GroupedNativeSelect = Template.bind({})
+GroupedNativeSelect.args = {
+  children: [
+    <NativeSelectOptionGroup key={0} label="Option Group 1">
+      <NativeSelectOption value="o-1" label="Option 1" />
+      <NativeSelectOption value="o-2" label="Option 2" />
+      <NativeSelectOption value="o-3" label="Option 3" />
+    </NativeSelectOptionGroup>,
+    <NativeSelectOptionGroup key={1} label="Option Group 2">
+      <NativeSelectOption value="o-4" label="Option 4" />
+      <NativeSelectOption value="o-5" label="Option 5" />
+    </NativeSelectOptionGroup>,
+  ],
+}
+
+export const DisabledNativeSelect = Template.bind({})
+DisabledNativeSelect.args = {
+  disabled: true,
+  children: [
+    <NativeSelectOption key={0} value="o-1" label="Option 1" />,
+    <NativeSelectOption key={1} value="o-2" label="Option 2" />,
+    <NativeSelectOption key={2} value="o-3" label="Option 3" />,
+  ],
+}
+
+export const InvalidNativeSelect = Template.bind({})
+InvalidNativeSelect.args = {
+  invalid: true,
+  children: [
+    <NativeSelectOption key={0} value="o-1" label="Option 1" />,
+    <NativeSelectOption key={1} value="o-2" label="Option 2" />,
+    <NativeSelectOption key={2} value="o-3" label="Option 3" />,
+  ],
+}
+
+export const ValidNativeSelect = Template.bind({})
+ValidNativeSelect.args = {
+  valid: true,
+  children: [
+    <NativeSelectOption key={0} value="o-1" label="Option 1" />,
+    <NativeSelectOption key={1} value="o-2" label="Option 2" />,
+    <NativeSelectOption key={2} value="o-3" label="Option 3" />,
+  ],
+}
+
+export const LoadingNativeSelect = Template.bind({})
+LoadingNativeSelect.args = {
+  loading: true,
+  children: [],
+}
+
+export const NativeSelectWithError = Template.bind({})
```

Comment:
Do we need to export all of them? I think to reuse them in other components right? also nice point of improvements like Epic for grouping them all together in one...